pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/PercentileRawEstAggregationFunction.java

```diff
@@ -0,0 +1,137 @@
+  public String getResultColumnName() {
+    final double percentile = _percentileRawEstAggregationFunction._percentile;
+    final int version = _percentileRawEstAggregationFunction._version;
+    final String type = getType().getName().toLowerCase();
+
+    return version == 0 ? type + (int) percentile + "(" + _expression + ")"
+        : type + "(" + _expression + ", " + percentile + ")";
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return _percentileRawEstAggregationFunction.createAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return _percentileRawEstAggregationFunction.createGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    _percentileRawEstAggregationFunction.aggregate(length, aggregationResultHolder, blockValSetMap);
+  }
+
+  @Override
+  public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    _percentileRawEstAggregationFunction.aggregateGroupBySV(length, groupKeyArray, groupByResultHolder, blockValSetMap);
+  }
+
+  @Override
+  public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    _percentileRawEstAggregationFunction
+        .aggregateGroupByMV(length, groupKeysArray, groupByResultHolder, blockValSetMap);
+  }
+
+  @Override
+  public QuantileDigest extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    return _percentileRawEstAggregationFunction.extractAggregationResult(aggregationResultHolder);
+  }
+
+  @Override
+  public QuantileDigest extractGroupByResult(GroupByResultHolder groupByResultHolder, int groupKey) {
+    return _percentileRawEstAggregationFunction.extractGroupByResult(groupByResultHolder, groupKey);
+  }
+
+  @Override
+  public QuantileDigest merge(QuantileDigest intermediateResult1, QuantileDigest intermediateResult2) {
+    return _percentileRawEstAggregationFunction.merge(intermediateResult1, intermediateResult2);
+  }
+
+  @Override
+  public boolean isIntermediateResultComparable() {
+    return _percentileRawEstAggregationFunction.isIntermediateResultComparable();
+  }
+
+  @Override
+  public ColumnDataType getIntermediateResultColumnType() {
+    return _percentileRawEstAggregationFunction.getIntermediateResultColumnType();
+  }
+
+  @Override
+  public ColumnDataType getFinalResultColumnType() {
+    return ColumnDataType.STRING;
+  }
+
+  @Override
+  public String extractFinalResult(QuantileDigest intermediateResult) {
+    return BytesUtils.toHexString(CustomSerDeUtils.QUANTILE_DIGEST_SER_DE.serialize(intermediateResult));
```

Comment:
Just talking to myself here - we have used this utility function in several places, and I can never remember its name (including ByteUtils). Perhaps we should make a Pinot Utils wrapper and use that. (Not a blocker for this PR).

Reply:
@mayankshriv Some of these serialized hex strings are little longer (about 3K bytes). Thinking of compressing this using [Snappy](https://github.com/xerial/snappy-java) or [LZ4](https://github.com/lz4/lz4-java). Did a quick test using LZ4 and got the following results. Pinot seems to [use](https://github.com/apache/pinot/blob/b826f2f588bc8d0d05ef466468c2317119ce46cc/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/compression/LZ4Compressor.java) LZ4 to compress data in segments already. 

Original Size = 3904; Compressed size = 1599; Compression rate = 59.04200819672132 

`  LZ4Factory factory = LZ4Factory.fastestInstance();
    LZ4Compressor compressor = factory.highCompressor(17);
    int maxCompressedLength = compressor.maxCompressedLength(decompressedLength);
    byte[] compressed = new byte[maxCompressedLength];
    int compressedSize = compressor.compress(data, 0, decompressedLength, compressed, 0, maxCompressedLength);
    double ratio = (1 - compressedSize / (double) originalSize) * 100.0;`

Was wondering if we should convert it back to base64 string or just return byte array (lz4 produces byte array). Let me know if you have any other ideas around compressing the results.

Reply:
We can do this outside this PR. Perhaps provide a wrapper to encode/decode these bytes, and share it in the client side library. Once we have a wrapper, we can do all sorts of things like compression etc. One reason why we did String was so UI can show it without breaking. Would have to evaluate that as well.

Reply:
Sounds good. Do we have a client side library ?

Reply:
We have the `pinot-client` library.

Reply:
Will look into it and add some compress logic if needed after this PR is landed.